services/collaboration/src/events/consumer.ts

```diff
@@ -0,0 +1,59 @@
+import { Queues } from './queues';
+import { getChannel } from './broker';
+import { createRoomWithQuestion } from '../controllers/roomController';
+import { sendMessageToQueue } from './producer';
+
+/**
+ * Function to initialize the room consumer
+ */
+export async function initializeRoomConsumer() {
+    try {
+        const channel = await getChannel();
+
+        channel.assertQueue(Queues.MATCH_FOUND, { durable: true });
+
+        channel.consume(Queues.MATCH_FOUND, async (msg: any) => {
+            if (msg) {
+                try {
+                    const content = JSON.parse(msg.content.toString());
+
+                    console.log(`Message consumed from MATCH_FOUND queue: ${JSON.stringify(content)}`);
+
+                    const { user1, user2, topics, difficulty } = content;
+
+                    const { requestId: requestId1 } = user1;
+                    const { requestId: requestId2 } = user2;
+
+                    const roomId = await createRoomWithQuestion(user1, user2, topics, difficulty);
+
+                    if (roomId) {
+                        console.log(`Room created successfully with ID: ${roomId}`);
+
+                        const collabCreatedMessage = {
+                            requestId1,
+                            requestId2,
+                            collabId: roomId,
+                        };
+
+                        console.log(
+                            `Message to be produced to COLLAB_CREATED queue: ${JSON.stringify(collabCreatedMessage)}`,
+                        );
+
+                        await sendMessageToQueue(Queues.COLLAB_CREATED, collabCreatedMessage);
+
+                        console.log(`Message sent to COLLAB_CREATED queue: ${JSON.stringify(collabCreatedMessage)}`);
+                    } else {
+                        console.log('Failed to create room.');
```

Comment:
We need to handle the case where the room was not created by producing a message on another queue (perhaps `Queues.COLLAB_CREATE_FAILED`) which is consumed by the match service so that it's status may be updated accordingly

Reply:
I have added the queue "COLLAB_CREATE_FAILED" on the collaboration side, and this queue will be populated with the message if the room is not created successfully.